drivers/ethernet/phy/phy_qualcomm_ar8031.c

```diff
@@ -0,0 +1,521 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * Inspiration from phy_mii.c, which is:
+ * Copyright (c) 2021 IP-Logix Inc.
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT qca_ar8031
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/net/mii.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(phy_qc_ar8031, CONFIG_PHY_LOG_LEVEL);
+
+#define AR8031_PHY_ID1           0x004DU
+#define PHY_READID_TIMEOUT_COUNT 1000U
+
+/* PHY Specific Status Register */
+#define PHY_SPECIFIC_STATUS_REG     0x11
+#define SPEC_STATUS_REG_LINK_MASK   (1U << 10)
+#define SPEC_STATUS_REG_DUPLEX_MASK (1U << 13)
+#define PHY_DUPLEX_HALF             (0U << 13)
+#define PHY_DUPLEX_FULL             (1U << 13)
+#define SPEC_STATUS_REG_SPEED_MASK  (0x3U << 14)
+#define PHY_SPEED_10M               (0U << 14)
+#define PHY_SPEED_100M              (1U << 14)
+#define PHY_SPEED_1000M             (2U << 14)
+
+/* The PHY Debug port address register */
+#define PHY_DEBUGPORT_ADDR_REG 0x1DU
+/* The PHY Debug port data register */
+#define PHY_DEBUGPORT_DATA_REG 0x1EU
+
+/* Debug port registers */
+/* Analog Test Control */
+#define PHY_DEBUGPORT_ANALOG_CTRL          0x0
+#define PHY_DEBUGPORT_ANALOG_CTRL_RX_DELAY (1U << 15)
+/* SerDes Test and System Mode Control */
+#define PHY_DEBUGPORT_SD_SM_CTRL           0x5
+#define PHY_DEBUGPORT_SD_SM_CTRL_TX_DELAY  (1U << 8)
+
+/* PHY MMD device address 3 */
+#define PHY_MMD_DEVICEID3              3U
+/* EEE capability Register */
+#define PHY_MMD3_EEE_CAP               0x14U
```

Comment:
pls add MMD generic registers to `include/zephyr/net/mdio.h`, so that other PHY drivers can reuse them. This avoid having the same defines on each PHY driver implementation. This comment apply for all other registers below.

Reply:
Updated, thanks.

Reply:
replied in https://github.com/zephyrproject-rtos/zephyr/pull/75073#discussion_r1706378210. I'm resolving this comment to keep the discussion in one thread.